type.go

```diff
@@ -106,7 +118,24 @@ func FromString(s string) Type {
 		return Timestamp
 	case Bool.String():
 		return Bool
+	case TNullString.String():
+		return TNullString
+	case TNullInt64.String():
+		return TNullInt64
+	case TNullFloat64.String():
+		return TNullFloat64
+	case TNullBool.String():
+		return TNullBool
 	default:
 		return Invalid
 	}
 }
+
+func isValidPrimaryKeyType(t Type) bool {
+	switch t {
+	case Invalid, TNullString, TNullFloat64, TNullInt64, TNullBool:
```

Comment:
This looks backwards. `isValidPrimaryKeyType` of `TNullFloat` returns `true` but `TNullFloat` is not a valid primary key type. Perhaps you meant `isInvalidPrimaryKeyType`?